src/components/motor_output_reordering/MotorOutputReorderingCanvas.js

```diff
@@ -0,0 +1,271 @@
+                this._ctx.closePath();
+                this._ctx.fillStyle = this._config.MotorReadyColor;
+                this._ctx.fill();
+            }
+
+            if (-1 !== mouseHoverMotorIndex && !this.readyMotors.includes(mouseHoverMotorIndex)) {
+                this._ctx.beginPath();
+                this._ctx.arc(motors[mouseHoverMotorIndex].x, motors[mouseHoverMotorIndex].y, this._config[this._droneConfiguration].PropRadius, 0, 2 * Math.PI);
+                this._ctx.closePath();
+                this._ctx.fillStyle = this._config.MotorMouseHoverColor;
+                this._ctx.fill();
+            }
+        } else {
+            const spinningMotor = this._motorIndexToSpinOnMouseDown;
+
+            for (let i = 0; i < motors.length; i++) {
+                this._ctx.fillStyle = this._config.MotorReadyColor;
+                if (spinningMotor === i) {
+                    this._ctx.fillStyle = this._config.MotorSpinningColor;
+                } else if (mouseHoverMotorIndex === i) {
+                    this._ctx.fillStyle = this._config.MotorMouseHoverColor;
+                }
+
+                this._ctx.beginPath();
+                this._ctx.arc(motors[i].x, motors[i].y, this._config[this._droneConfiguration].PropRadius, 0, 2 * Math.PI);
+                this._ctx.closePath();
+                this._ctx.fill();
+            }
+        }
+    }
+
+    _getMouseHoverMotorIndex()
+    {
+        const x = this._mouse.x;
+        const y = this._mouse.y;
+
+        let result = -1;
+        let currentDist = Number.MAX_SAFE_INTEGER;
+        const motors = this._config[this._droneConfiguration].Motors;
+
+        for (let i = 0; i < motors.length; i++) {
+            const dist = Math.sqrt((x - motors[i].x) * (x - motors[i].x) + (y - motors[i].y) * (y - motors[i].y));
+
+            if (dist < this._config[this._droneConfiguration].PropRadius && dist < currentDist) {
+                currentDist = dist;
+                result = i;
+            }
+        }
+
+        return result;
+    }
+
+    _drawMotors()
+    {
+
+        this._ctx.lineWidth = this._config.PropEdgeLineWidth;
+        this._ctx.strokeStyle = this._config.PropEdgeColor;
+        const motors = this._config[this._droneConfiguration].Motors;
+
+        for (let i = 0; i < motors.length; i++) {
+            this._ctx.beginPath();
+            this._ctx.arc(motors[i].x, motors[i].y, this._config[this._droneConfiguration].PropRadius, 0, 2 * Math.PI);
+            this._ctx.stroke();
+
+            /*
+            // uncomment for checking new configurations motor order
+            this._ctx.font = this._config.MotorNumberTextFont;
+            this._ctx.fillStyle = this._config.MotorNumberTextColor;
+            this._ctx.textAlign = "center";
+            this._ctx.textBaseline = "middle";
+            this._ctx.fillText(i + 1, motors[i].x, motors[i].y);
+            */
```

Comment:
As we have commented, we don't like to have commented code. Let's see if others developers can suggest something better...

Reply:
@McGiverGim so the purpose of this commented code is this one:
when we need to add more quad configurations, it's important that the motors will be programmed to render in the correct order/placews. With this code, it's easy to just quickly check if all the motors are placed in the correct spots.